src/widget/wnumberpos.cpp

```diff
@@ -45,73 +45,97 @@ WNumberPos::WNumberPos(const char* group, QWidget* parent)
+        slotSetPosition(m_dOldPosition);
     }
 }
 
 void WNumberPos::slotSetTrackSamples(double dSamples) {
     m_dTrackSamples = dSamples;
-    slotSetValue(m_dOldValue);
+    slotSetPosition(m_dOldPosition);
 }
 
 void WNumberPos::slotSetTrackSampleRate(double dSampleRate) {
     m_dTrackSampleRate = dSampleRate;
-    slotSetValue(m_dOldValue);
+    slotSetPosition(m_dOldPosition);
 }
 
+// Reimplementing WNumber::setValue
 void WNumberPos::setValue(double dValue) {
     // Ignore midi-scaled signals from the skin connection.
     Q_UNUSED(dValue);
     // Update our value with the old value.
-    slotSetValue(m_dOldValue);
+    slotSetPosition(m_dOldPosition);
 }
 
-void WNumberPos::slotSetValue(double dValue) {
-    m_dOldValue = dValue;
+void WNumberPos::slotSetPosition(double dPosition) {
+    m_dOldPosition = dPosition;
 
-    double dPosSeconds = 0.0;
+    double dPosSecondsElapsed = 0.0;
+    double dPosSecondsRemaining = 0.0;
     if (m_dTrackSamples > 0 && m_dTrackSampleRate > 0) {
         double dDurationSeconds = (m_dTrackSamples / 2.0) / m_dTrackSampleRate;
         double dDurationMillis = dDurationSeconds * 1000.0;
-        double dPosMillis = dValue * dDurationMillis;
-        if (m_bRemain) {
-            dPosMillis = math_max(dDurationMillis - dPosMillis, 0.0);
+        double dPosMillis = dPosition * dDurationMillis;
+        dPosSecondsElapsed = dPosMillis / 1000.0;
+        if (m_displayMode != DisplayMode::Elapsed) {
+            double dPosMillisRemaining = math_max(dDurationMillis - dPosMillis, 0.0);
+            dPosSecondsRemaining = dPosMillisRemaining / 1000.0;
         }
-        dPosSeconds = dPosMillis / 1000.0;
     }
 
-    QString sPosText;
-    if (dPosSeconds >= 0.0) {
-        sPosText = m_skinText % mixxx::Duration::formatSeconds(
-                dPosSeconds, mixxx::Duration::Precision::CENTISECONDS);
-    } else {
-        sPosText = m_skinText % QLatin1String("-") % mixxx::Duration::formatSeconds(
-                -dPosSeconds, mixxx::Duration::Precision::CENTISECONDS);
+    if (m_displayMode == DisplayMode::Elapsed) {
+        if (dPosSecondsElapsed >= 0.0) {
+            setText(mixxx::Duration::formatSeconds(
+                        dPosSecondsElapsed, mixxx::Duration::Precision::CENTISECONDS));
+        } else {
+            setText(QLatin1String("-") % mixxx::Duration::formatSeconds(
+                        -dPosSecondsElapsed, mixxx::Duration::Precision::CENTISECONDS));
+        }
+    } else if (m_displayMode == DisplayMode::Remaining) {
+        setText(QLatin1String("-") % mixxx::Duration::formatSeconds(
+                    dPosSecondsRemaining, mixxx::Duration::Precision::CENTISECONDS));
+    } else if (m_displayMode == DisplayMode::ElapsedAndRemaining) {
+        if (dPosSecondsElapsed >= 0.0) {
+            setText(mixxx::Duration::formatSeconds(
+                        dPosSecondsElapsed, mixxx::Duration::Precision::CENTISECONDS)
+                    % QLatin1String(" / -") %
```

Comment:
To me the ``/`` separator is unneeded.
Depending on the font, i found it harder to read with it, we already have the ``-``in front of the remaining time display as separator, so 3 spaces and ``-`` should do it.